src/EditorFeatures/CSharpTest/Completion/CompletionProviders/PartialTypeCompletionProviderTests.cs

```diff
@@ -0,0 +1,291 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Completion;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.CodeAnalysis.CSharp.Completion.Providers;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Completion.CompletionProviders
+{
+    public class PartialTypeCompletionProviderTests : AbstractCSharpCompletionProviderTests
+    {
+        public PartialTypeCompletionProviderTests(CSharpTestWorkspaceFixture workspaceFixture) : base(workspaceFixture)
+        {
+        }
+
+        internal override CompletionProvider CreateCompletionProvider()
+        {
+            return new PartialTypeCompletionProvider();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task TestRecommendTypesWithoutPartial()
+        {
+            var text = @"
+class C { }
+
+partial class $$";
+
+            await VerifyItemIsAbsentAsync(text, "C");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task TestPartialClass1()
+        {
+            var text = @"
+partial class C { }
+
+partial class $$";
+
+            await VerifyItemExistsAsync(text, "C");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task TestPartialGenericClass1()
+        {
+            var text = @"
+class Bar { }
+
+partial class C<Bar> { }
+
+partial class $$";
+
+            await VerifyItemExistsAsync(text, "C<Bar>");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task TestPartialGenericClassCommitOnTab()
+        {
+            var text = @"
+class Bar { }
+
+partial class C<Bar> { }
+
+partial class $$";
+
+            var expected = @"
+class Bar { }
+
+partial class C<Bar> { }
+
+partial class C<Bar>";
+
+            await VerifyProviderCommitAsync(text, "C<Bar>", expected, null, "");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task TestPartialGenericClassCommitOnSpace()
+        {
+            var text = @"
+partial class C<T> { }
+
+partial class $$";
+
+            var expected = @"
+partial class C<T> { }
+
+partial class C<T> ";
+
+            await VerifyProviderCommitAsync(text, "C<T>", expected, ' ', "");
```

Comment:
I think the background here is that the C# symbol provider doesn't ever commit the `<`.  This is because it never really knows any instantiation info, so it can't commit anything useful except the `<` itself, which is just one character and so didn't warrant it.

Providers that *do* know instantiation info (this, `ObjectCreationCompletionProvider`, the cref provider, etc) don't tend to handle the overtyping case well in C#.  We could definitely change that, but I don't think it needs to be part of this PR.